### DISCUSSION TITLE
272

### DISCUSSION THREAD

**Discussion by jojoldu** [jojoldu/blog-comments#696] [Comments]
http://jojoldu.tistory.com/272

**Comment by jojoldu** [comment]
작성자: soulduse@naver.com

좋은 자료 이렇게 만들어 주셔서 감사합니다!
고생하셨습니다.!

**Comment by necronia** [comment]
꼭 필요했고
너무나 도움이 많이 되는 강좌였습니다
진심으로 감사드립니다^^

**Comment by jojoldu** [comment]
@necronia 좋게 봐주셔서 정말 감사드립니다^^

**Comment by rlackdfuf200** [comment]
너무 수고 많으셨고 좋은 자료 감사합니다~

**Comment by orgin1127** [comment]
기억 나시련지 모르겠지만 너무나 큰 도움을 받아서 감사인사 남깁니다. 덕분에 목표했던 웹 어플리케이션을 구현했습니다. https://github.com/orgin1127/terraceHouse 깃헙 주소구요 코딩 8개월차의 아직 많이 부족한 초보지만 이정도까지 할 수 있었던건 정말 이 포스팅 시리즈 덕입니다. 
초보라 실수도 많이 하고 에러 터지면 붙잡고 스택 오버플로우 뒤지면서 어찌어찌 해결하고 그랬지만, 가장 많이 도움 받은건 이 시리즈 포스팅과 JPA 관련 포스팅이었습니다. 몇번을 봤는지... 
정말 자세하게 기초적인 내용을 잘 설명해주시고 예시도 잘 써주셔서 덕분에 그걸 가지고 많은 부분을 활용할수 있었습니다.
정말 이렇게 좋은 자료 남겨주셔서 감사드립니다.  ^^

**Comment by jojoldu** [comment]
@orgin1127 와!! 멋지십니다! 제가 8개월일때는 이정도 퀄리티로 전혀 만들지 못했던걸 생각해보면 정말 대단하신것 같습니다 ㅎㅎ 
잘 봐주셔서 제가 더 감사할 따름입니다 :)
블로그할 힘이 나네요 ^^
계속 글 쓰겠습니다
고맙습니다 ^^

**Comment by jhleon86** [comment]
드디어 완주했네요..^^
이제 도화지에 색연필까지 준비가 되었으니 무엇을 그려볼까 행복한 고민을 또다시 시작합니다. 
덕분에 여기까지 왔습니다 
앞으로도 자주와서 많이 배워가겠습니다.
감사힙니다~~

**Comment by gonghojin** [comment]
안녕하세요. 창천항로님 질문하나만 드려도 될까요...??
스프링부트 서비스글 완주해서 페이스북에서 추천해주신 마x님 블로그 적용해보고 있는데요.
창천항로님 글 베이스에 mockito 테스트를 하려면 별도의 설정이 필요한가요??.
postman으로 값을 넘기거나 service층 테스트 할 경우 문제없이 정상작동하는데
아래의 mockito 메소드를 사용할 경우 같은 로직을 걸쳐도 null 값이 뜨네요.


@RunWith(SpringRunner.class)
@SpringBootTest
@AutoConfigureMockMvc
public class PostControllerTest {

    @Autowired
    private MockMvc mvc;

    @MockBean
    private PostsService postsService;

    // TODO: postman이나 service 테스트할 경우 null 에러가 뜨지 않으나, mockito를 거치면 에러가 존재- 해결 필요
    @Test
    public void 포스트_등록() throws Exception {
       /* Posts entity = PostsSaveRequestDto.builder()
                .author("gongdel@gmail.com")
                .content("테스트")
                .title("테스트 타이틀")
                .code("마크업")
                .build().toEntity();
        given(postsService.save(any())).willReturn(entity);*/

        this.mvc.perform(post("/posts/new")
                .param("title", "제목1")
                .param("content", "내용1")
                .param("author", "gongdel")
                .param("code", "마크다운1"))
                .andExpect(status().isFound())
                .andExpect(header().string(HttpHeaders.LOCATION, "/posts/1"));
    }

이 거 말고는 dependencies에서 testCompile "org.mockito:mockito-core:2.+" 부분 추가해준 거 말고는 별도의 설정은 없습니다.(나머지 설정은 스프링부트 글 베이스)
디버깅해보면 invocationimpl 클래스에
Method threw 'java.lang.NullPointerException' exception. Cannot evaluate org.mockito.internal.invocation.InvocationImpl.toString() 메시지가 뜨네요. 
(참고 소스가 부족하면 github에 커밋되있습니다. 자주 질문드려서 죄송합니다~)

**Comment by Y-k-Y** [comment]
좋은 자료 정말 감사드립니다. 혼자 공부하고 있는 중 아주 큰 힘이 되었습니다.🙇🏻

**Comment by newzinee** [comment]
정말 감사드립니다.

**Comment by kornetbob** [comment]
좋은글 감사합니다.
작년에 작성하셨던거 정독은 오늘했네요^^
현재 님이 말씀대로 제가 서비스를 못한 이유가 다 적혀있네요.ㅎ
현직에서는 프론트 담당이고, 개인적으로 자바 스프링을 공부하는데, 사람을 유지시켜야된다의 마인드만 커서, 새로운 컨텐츠, 추가추가 하다가 재작년에 손을 놔버렸어요.ㅎ

다시 시도하려고, 하지만 쉽지는 않는데, 기초 템플릿 작업하면서 작성해놓은게 있어 그 기반으로 다시 시도 하고 있습니다.

제 이야기이고, 필요한 내용들이 많아서 도움이 많이 되었습니다. 정말 감사합니다.^^

**Comment by kc7851** [comment]
정말 좋은 글을 포스팅 해주셔서 감사합니다.
최고입니다!!!

**Comment by NotRayor** [comment]
잘보고 갑니다

**Comment by CockPc** [comment]
탈무드에 적힐만한 글입니다
